### PR TITLE
Replace ForecastDFDataset + Trainer.predict() with proper inference paths in TTMForecaster

### PR DESCRIPTION
**Pull request by ShivamSingal**
## Problem

`TTMForecaster.predict()` and `predict_zero_shot()` used `ForecastDFDataset` + `Trainer.predict()` for inference. This is incorrect — `ForecastDFDataset` is a **training abstraction** that creates supervised `(context, future_values)` window pairs, and expects `context_length + prediction_length` rows per time series.

When used for inference (where only context exists, no future ground truth):

- **`enable_padding=True` (the default):** The dataset zero-pads by **prepending** rows to reach the required length. With 512 context rows and 72 forecast horizon, it needs 584 rows, so it prepends 72 zeros. The model then receives `[72 zeros | 440 real rows]` as context instead of the full 512 real rows — **silently corrupting the input tensor**.
- **`enable_padding=False`:** The dataset correctly rejects the short input, producing 0 samples. Inference returns nothing.

Neither behavior is correct. The root cause is using a training dataset class for inference.

## Evidence from IBM granite-tsfm maintainers

IBM built a dedicated inference class (`TimeSeriesForecastingPipeline`) specifically for this use case. The lead maintainer `@wgifford` clarified the intended separation across multiple GitHub issues:

- **[ibm-granite/granite-tsfm#53](https://github.com/ibm-granite/granite-tsfm/issues/53):** *"The TimeSeriesForecastingPipeline is intended for inference."*
- **[ibm-granite/granite-tsfm#172](https://github.com/ibm-granite/granite-tsfm/issues/172):** *"trainer.predict works with torch datasets and outputs tensors. For higher-level functionality, we created the TimeSeriesForecastingPipeline. The pipeline takes dataframes as input and outputs dataframes."*
- **[ibm-granite/granite-tsfm#363](https://github.com/ibm-granite/granite-tsfm/issues/363):** *"the preferred path is to create forecasts in the original unscaled domain using the time series forecasting pipeline."*
- **[ibm-granite/granite-tsfm#46](https://github.com/ibm-granite/granite-tsfm/issues/46):** Users reported the exact same confusion about how to run inference; `@wgifford` pointed them to the pipeline.

The pipeline internally wraps `ForecastDFDataset` correctly — it calls `extend_time_series()` to pad with NaN-filled future rows before creating the dataset, which is the proper way to handle context-only input.

For zero-shot inference specifically, IBM's documentation states users should *"externally standard scale their data independently for every channel before feeding it to the model"* — fitting a global `TimeSeriesPreprocessor` across all test patients would reintroduce distribution shift.

### `predict()` — fine-tuned inference
Replaced `ForecastDFDataset` + `Trainer.predict()` with `TimeSeriesForecastingPipeline`:
```python
pipeline = TimeSeriesForecastingPipeline(
    model=self.model,
    feature_extractor=self.preprocessor,
    explode_forecasts=True,
    inverse_scale_outputs=inverse_scale,
)
forecast_df = pipeline(data)
```
The pipeline handles scaling → tensor construction → model forward pass → inverse scaling, returning a DataFrame in original units.

### `predict_zero_shot()` — zero-shot inference
Replaced `ForecastDFDataset` + `Trainer.predict()` with `TimeSeriesForecastingPipeline` **without** a `feature_extractor`:
```python
pipeline = TimeSeriesForecastingPipeline(
    model=self.model,
    timestamp_column=...,
    id_columns=...,
    target_columns=...,
    observable_columns=...,
    explode_forecasts=True,
    freq=f"{self.config.resolution_min}min",
)
forecast_df = pipeline(data)
```
Without a fitted preprocessor, TTM's internal `TinyTimeMixerStdScaler` (RevIN) handles per-window standardization automatically during the forward pass — no manual scaling or global scaler leakage.

### Dead code removed
- `_prepare_inference_data()` — created `ForecastDFDataset` + `DataLoader` for inference
- `_create_inference_trainer()` — created `Trainer` with `label_names` workaround
- `_inverse_scale_zero_shot_predictions()` — was never called from anywhere

### Why backbone embeddings indexing is gone
The old code used `Trainer.predict()` which returns a `TinyTimeMixerForPredictionOutput` flattened to a tuple: `predictions[0]` = forecast tensor, `predictions[1]` = backbone hidden state (`(B, C_input, num_patches, d_model)`). The old code correctly extracted these, but we only need forecasts — not encoder representations. With `TimeSeriesForecastingPipeline`, the pipeline returns a DataFrame of forecasts directly, so there's no tuple indexing or backbone embeddings to deal with.

### Unchanged
- **Training path** (`_prepare_training_data` + `Trainer.fit`): `ForecastDFDataset` is correct here — training needs supervised (context, future) pairs
- `_inverse_scale_predictions()`: retained (referenced by existing test)
- `_save_checkpoint()` / `_load_checkpoint()`: unchanged

## Architecture after this PR

| Code path | Method | Why |
|---|---|---|
| Training / fine-tuning | `ForecastDFDataset` + `Trainer` | Training needs supervised (context, future) pairs |
| Fine-tuned inference | `TimeSeriesForecastingPipeline` | Official IBM inference path, uses fitted preprocessor |
| Zero-shot inference | `TimeSeriesForecastingPipeline` (no preprocessor) | TTM's internal RevIN scaler handles per-window normalization |

## Validation (watgpu, Brown 2019 holdout, midnight-anchored episodes)

| Path | RMSE (mM) | Discontinuity (mM) | Episodes | Job ID |
|---|---|---|---|---|
| Fine-tuned | 2.006 ± 1.438 | 0.198 ± 0.183 | 1830 | 1407002 |
| Zero-shot | 2.469 ± 1.345 | 0.210 ± 0.148 | 100 | 1407018 |

Fine-tuned results are an **exact match** (diff = 0.000) against the known-good direct-inference baseline from the scaler experiment (Job 1406872). Zero-shot discontinuity (0.210 mM) is well within acceptable range.

Net change: **+32 / -356 lines**, single file (`src/models/ttm/model.py`).

## Test plan
- [x] Fine-tuned validation: RMSE=2.006, Disc=0.198, 1830 episodes, 0 errors
- [x] Zero-shot validation: RMSE=2.469, Disc=0.210, 100 episodes, 0 errors
- [ ] `tests/models/test_ttm_preprocessor_roundtrip.py` (uses retained `_inverse_scale_predictions`)

### CODE REVIEW

**Review by RobotPsychologist — APPROVED** [review]
LOOKS GOOD TO ME
